src/implementations/utils/wallet.ts

```diff
@@ -21,33 +36,51 @@ export const getPrivateKey = ({ keyFile, key }: WalletOption): string | undefine
   }
   return key || getKeyFromFile(keyFile) || process.env["OA_PRIVATE_KEY"];
 };
-export const getWallet = async ({
-  keyFile,
-  key,
+
+export const getWalletOrSigner = async ({
   network,
-  encryptedWalletPath,
   progress = defaultProgress("Decrypting Wallet"),
-}: WalletOption & Partial<NetworkOption> & { progress?: (progress: number) => void }): Promise<Wallet> => {
+  ...options
+}: WalletOrSignerOption & Partial<NetworkOption> & { progress?: (progress: number) => void }): Promise<
+  Wallet | ConnectedSigner
+> => {
   const provider =
     network === "local"
       ? new providers.JsonRpcProvider()
       : getDefaultProvider(network === "mainnet" ? "homestead" : network); // homestead => aka mainnet
-  if (encryptedWalletPath) {
+  if (isWalletOption(options)) {
```

Comment:
would a switch case for the logic branches be better here instead of multiple if-else blocks?